elasticdl/python/master/master.py

```diff
@@ -184,6 +190,8 @@ def run(self):
         try:
             while True:
                 if self.task_d.finished():
+                    for callback in self.callbacks:
+                        callback.on_train_end()
```

Comment:
From this code, this callback is executed on master pod.
How about callback to upload the exported model to remote storage, do master or worker execute it?
We may consider that the model is exported by the worker.

Reply:
"How about callback to upload the exported model to remote storage, do master or worker execute it?"
The master will tar the model and upload.

Now, the model is exported from checkpoints by the worker.  If we want to execute callbacks with fault-tolerant,  the master should create a task for each callback on the epoch end and train/test end.

Reply:
I assume you means that the model is exported by the worker, and then the master will compress and upload the model to remote storage.
I'm wondering how master access the exported model from worker? Maybe shared storage in glusterfs?

How about letting one single pod do the model export and upload work?

Reply:
I have removed the code about callback execution on the master and only remain the callback definition. The callback execution needs further consideration for fault-tolerant.

Reply:
> I assume you mean that the model is exported by the worker, and then the master will compress and upload the model to the remote storage.
> I'm wondering how master accesses the exported model from worker? Maybe shared storage in glusterfs?
> 
> How about letting one single pod do the model export and upload work? We can compare these two options.

The model is exported to glusterfs by the worker and the `save_model_path` is sent to the worker by the master. So the master can access the exported model by `save_model_path`. Before, we assign a saved_model task to a worker because we think the memory of a worker is enough to load a model.  I feel like it is complicated to assign a single pod to execute the method of the callback and some callback must be called by the master. Then we can discuss the problem about calling the methods of callback in  #1706.